### ISSUE TITLE
[BUG] Cannot dump from a database which is accessed with a password containing a single-quote

### ISSUE THREAD

**Issue by roger-duthie-pivotal** [cweijan/vscode-mysql#1278]
- Database type and version: PostgreSQL 14

Steps:
1. Connect PostgreSQL database in the client, with a user which has a password containing a single-quote
2. Right-click on the connected DB and selecting Dump Data & Structure

The `psql` command used "under the bonnet" fails because the password is enclosed by single-quotes.

```shell
export PGPASSWORD='PASSWORD'CONTAINING'SINGLE'QUOTES' && pg_dump -h 127.0.0.1 -w -p 5432 -U [USER_NAME] [DB_NAME] > [OUTPUT_FILE]
```

Perhaps this could be easily fixed by checking whether single-quotes or double-quotes are in the password and using the appropriate enclosing characters or using escape sequences?

For example:

```shell
export PGPASSWORD="PASSWORD'CONTAINING'SINGLE'QUOTES" && pg_dump -h 127.0.0.1 -w -p 5432 -U [USER_NAME] [DB_NAME] > [OUTPUT_FILE]
```

... would work for passwords with single-quotes and more generally, there are also double quotes:

```shell
export PGPASSWORD="PASSWORD'CONTAINING'SINGLE'QUOTES\"AND\"DOUBLE\"QUOTES" && pg_dump -h 127.0.0.1 -w -p 5432 -U [USER_NAME] [DB_NAME] > [OUTPUT_FILE]
```

**Comment by cweijan** [comment]
I will try to fix this, currently you can do it with a custom terminal command.

<img width="897" alt="图像" src="https://github.com/user-attachments/assets/e9e4f1c6-a75c-4476-a261-6967c8090cf1"><img width="713" alt="image" src="https://github.com/user-attachments/assets/e0680c5b-bf39-4a2a-9b41-3b42df0db816">

**Comment by cweijan** [comment]
Thanks for the feedback, it has been fixed in version 7.6.3.
